UnitTests/Sources/StartChatViewModelTests.swift

```diff
@@ -44,6 +45,43 @@ class StartChatScreenViewModelTests: XCTestCase {
         XCTAssertTrue(userDiscoveryService.searchProfilesWithCalled)
     }
     
+    func testJoinRoomByAddress() async throws {
+        clientProxy.resolveRoomAliasReturnValue = .success(.init(roomId: "id", servers: []))
+        viewModel.context.roomAddress = "#room:example.com"
+        
+        let deferredViewState = deferFulfillment(viewModel.context.$viewState) { viewState in
+            viewState.joinByAddressState == .addressFound(address: "#room:example.com", roomID: "id")
+        }
+        try await deferredViewState.fulfill()
```

Comment:
The order here seems weird, does this still work?
```suggestion
        
        let deferredViewState = deferFulfillment(viewModel.context.$viewState) { viewState in
            viewState.joinByAddressState == .addressFound(address: "#room:example.com", roomID: "id")
        }
        viewModel.context.roomAddress = "#room:example.com"
        try await deferredViewState.fulfill()
```
```suggestion
        viewModel.context.roomAddress = "#room:example.com"
        
        let deferredViewState = deferFulfillment(viewModel.context.$viewState) { viewState in
            viewState.joinByAddressState == .addressFound(address: "#room:example.com", roomID: "id")
        }
        try await deferredViewState.fulfill()
```